kuma/javascript/src/banners.jsx

```diff
@@ -114,43 +114,53 @@ export type BannerProps = {
     // name of the waffle flag that controls the banner, and is also
     // used as part of a localStorage key.
     id: string,
+    // class name used on main banner container. Exclusively used
+    // for styling purposes.
+    classname: string,
     // The banner title. e.g. "MDN Survey"
     title: string,
     // The banner description. e.g. "Help us understand the top 10 needs..."
-    copy: string,
+    // Could also be a React Element such as that returned by `<Interpolated />`
+    copy: Object | string,
     // The call to action button text. e.g. "Take the survey"
     cta: string,
     // The URL of the page to open when the button is clicked
     url: string,
+    // An optional property. If present, it should be set to true to indicate
+    // that this banner is to be shown to authenticated users only
+    authenticated?: boolean,
     // An optional property. If present, it specifies the number of days
     // for which a dismissed banner will not be shown. If omitted, the
     // default is 5 days.
-    embargoDays?: number
+    embargoDays?: number,
+    // An optional property. If present, it should be set to true to indicate
+    // that the main cta link should open in a new window
+    newWindow?: boolean
 };
 
 function Banner(props: BannerProps) {
     const [isDismissed, setDismissed] = useState(false);
+    const containerClassNames = `${props.classname} mdn-cta-container cta-background-linear`;
 
     if (isDismissed) {
         return null;
     }
 
     return (
-        <div className="mdn-cta-container">
-            <div
-                id="mdn-cta-content"
-                className="mdn-cta-content cta-background-linear"
-            >
-                <div id="developer-needs" className="mdn-cta-content-container">
+        <div className={containerClassNames}>
+            <div id="mdn-cta-content" className="mdn-cta-content">
+                <div id={props.id} className="mdn-cta-content-container">
                     <h2 className="mdn-cta-title slab-text">{props.title}</h2>
                     <p className="mdn-cta-copy">{props.copy}</p>
                 </div>
                 <p className="mdn-cta-button-container">
                     <a
                         href={props.url}
                         className="mdn-cta-button"
-                        target="_blank"
-                        rel="noopener noreferrer"
+                        {...(props.newWindow && {
```

Comment:
@Gregoor Flow is going to complain about this. Not sure if there is a better way to accomplish this that Flow will also be happy with or, whether we should quiet Flow for this one.

I tried various approaches. Changing the prop definition to `Boolean` removes the error here but then introduces it further down at line 112. Changing that line to be say `Boolean(true)` moves the error to line 258 :)